### ISSUE TITLE
[BUG] Volume fails to reattach when the node is powered down and the pod migrates

### ISSUE THREAD

**Issue by sowmyav27** [longhorn/longhorn#1307]
**Describe the bug**
Volume fails to reattach when the node is powered down and the pod migrates

**To Reproduce**

- Attach a volume to a pod on node 1
- Power down node 1
- The pod/workload goes into "Unknown" state when seen from Rancher UI.
- The pod migrates to another node
- But the volume fails to reattach. Error: `Multi-Attach error for volume "vol1" Volume is already used by pod(s) wk01-5669c54dd5-wxr4`

**Expected behavior**
Volume should be able to reattach to the pod on another node

**Environment:**
 - Longhorn version: master

**Comment by yasker** [comment]
It's #820 . After #820 is solved, this should be good.

The pod must be a part of deployment for #820 to work.

**Comment by yasker** [comment]
Duplicate of #820
